### PR TITLE
enhancement(kubernetes platform): Adds cue docs for k8s pod_ip and pod_ips annotations

### PR DESCRIPTION
**Pull request by eeyun**
Signed-off-by: Ian Henry <ianjhenry00@gmail.com>

### CODE REVIEW

File: docs/reference/components/sources/kubernetes_logs.cue [hunk 1]
```diff
@@ -77,6 +77,22 @@ components: sources: kubernetes_logs: {
 							default: "kubernetes.container_name"
 						}
 					}
+					pod_ip: {
+						common: false
+						description: "Event field for Pod IP."
```
Comment (lucperkins) [inline docs/reference/components/sources/kubernetes_logs.cue:82]: I think descriptions could afford to be a bit more descriptive. Maybe something like this:

"The name of the event field displaying the Pod's IP address."

Comment (eeyun) [reply docs/reference/components/sources/kubernetes_logs.cue]: I've clarified these slightly, but if we change these to make them more verbose we probably want to change the others as well. This is terse, agreed, but it follows the previously established pattern in the file. I'd personally prefer to get these in and then follow up syntactical changes with a secondary PR across the full file.

File: docs/reference/components/sources/kubernetes_logs.cue [hunk 2]
```diff
@@ -197,6 +213,24 @@ components: sources: kubernetes_logs: {
 					default: null
 				}
 			}
+			"kubernetes.pod_ip": {
+				description: "Pod labels IP address."
```
Comment (MOZGIII) [inline docs/reference/components/sources/kubernetes_logs.cue:217]:
```suggestion
				description: "Pod IP address."
```

File: docs/reference/components/sources/kubernetes_logs.cue [hunk 3]
```diff
@@ -197,6 +213,24 @@ components: sources: kubernetes_logs: {
...
+				required:    false
+				common:      true
+				type: string: {
+					examples: ["192.168.1.1"]
+					default: null
+				}
+			}
+			"kubernetes.pod_ips": {
+				description: "Pod labels IPv4 and IPv6 addresses."
```
Comment (MOZGIII) [inline docs/reference/components/sources/kubernetes_logs.cue:226]:
```suggestion
				description: "Pod IPv4 and IPv6 addresses."
```

File: docs/reference/components/sources/kubernetes_logs.cue [hunk 4]
```diff
@@ -274,6 +308,8 @@ components: sources: kubernetes_logs: {
 				"file":                       "/var/log/pods/kube-system_storage-provisioner_93bde4d0-9731-4785-a80e-cd27ba8ad7c2/storage-provisioner/1.log"
 				"kubernetes.container_image": "gcr.io/k8s-minikube/storage-provisioner:v3"
 				"kubernetes.container_name":  "storage-provisioner"
+				"kubernetes.pod_ip": "192.168.1.1"
+				"kubernetes.pod_ips": ["192.168.1.1", "::1"]
```
Comment (MOZGIII) [inline docs/reference/components/sources/kubernetes_logs.cue:312]: Looks like you might need to run `cue fmt`

File: docs/reference/components/sources/kubernetes_logs.cue (hunk 1 above)
Comment (MOZGIII) [reply docs/reference/components/sources/kubernetes_logs.cue]: I'm pretty sure I tried something in the form of `The name of the event field displaying the Pod's IP address.` first, and then after some corrections during review, we landed on what we have now at the surrounding fields.

Let's just do things consistently with the rest of the fields in this PR, and create an issue to revisit and update all of the fields afterwards, after a proper discussion? I feel like if we were to change the format - we should do it at once for all of the fields, and this shouldn't be in this PR.

**Review by MOZGIII — APPROVED** [review]

**Review by MOZGIII — APPROVED** [review]
Left some nits, overall looks good!

**Pull request merged**